plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/TableStatisticsMaker.java

```diff
@@ -124,6 +126,11 @@ private TableStatistics makeTableStatistics(IcebergTableHandle tableHandle)
             if (summary.getColumnSizes() != null) {
                 Long columnSize = summary.getColumnSizes().get(fieldId);
                 if (columnSize != null) {
+                    if (columnHandle.getBaseType().equals(VarcharType.VARCHAR) || columnHandle.getBaseType().equals(VarbinaryType.VARBINARY)) {
+                        // columnSize value is in fact size of column stored on disk which is after compression and is much smaller than
+                        // the column size in memory. Multiplying by 10 seems like a good heuristic to compensate for that
+                        columnSize = columnSize * 10;
```

Comment:
- statistical compression factor for binary is likely different than for text.
  i don't know whether we have any "data" to back up the `* 10` number
- when Iceberg type is `fixed(L)` the Trino type is `VARBINARY`, but Iceberg actually knows that each cell is `L` bytes. Can we tell the engine about that?